pkg/reconciler/tenancy/workspacemounts/workspacemounts_controller_workspaces.go

```diff
@@ -0,0 +1,310 @@
+	kcpclientset "github.com/kcp-dev/kcp/sdk/client/clientset/versioned/cluster"
+	tenancyv1alpha1client "github.com/kcp-dev/kcp/sdk/client/clientset/versioned/typed/tenancy/v1alpha1"
+	corev1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/core/v1alpha1"
+	tenancyv1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/tenancy/v1alpha1"
+	corev1alpha1listers "github.com/kcp-dev/kcp/sdk/client/listers/core/v1alpha1"
+	tenancyv1alpha1listers "github.com/kcp-dev/kcp/sdk/client/listers/tenancy/v1alpha1"
+)
+
+const (
+	// ControllerName is the name of this controller.
+	ControllerName = "kcp-workspace-mounts"
+)
+
+// NewController creates a new controller for generic mounts.
+func NewController(
+	shardName string,
+	kcpClusterClient kcpclientset.ClusterInterface,
+	kubeClusterClient kubernetes.ClusterInterface,
+	logicalClusterAdminConfig *rest.Config,
+	externalLogicalClusterAdminConfig *rest.Config,
+	dynamicClusterClient kcpdynamic.ClusterInterface,
+	workspaceInformer tenancyv1alpha1informers.WorkspaceClusterInformer,
+	globalShardInformer corev1alpha1informers.ShardClusterInformer,
+	globalWorkspaceTypeInformer tenancyv1alpha1informers.WorkspaceTypeClusterInformer,
+	logicalClusterInformer corev1alpha1informers.LogicalClusterClusterInformer,
+	discoveringDynamicSharedInformerFactory *informer.DiscoveringDynamicSharedInformerFactory,
+) (*Controller, error) {
+	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
+	gvkQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
+
+	logger := logging.WithReconciler(klog.Background(), ControllerName)
+
+	c := &Controller{
+		queue:    queue,
+		gvrQueue: gvkQueue,
+
+		shardName:                         shardName,
+		logicalClusterAdminConfig:         logicalClusterAdminConfig,
+		externalLogicalClusterAdminConfig: externalLogicalClusterAdminConfig,
+
+		kcpClusterClient:     kcpClusterClient,
+		kubeClusterClient:    kubeClusterClient,
+		dynamicClusterClient: dynamicClusterClient,
+
+		workspaceIndexer: workspaceInformer.Informer().GetIndexer(),
+		workspaceLister:  workspaceInformer.Lister(),
+
+		globalShardIndexer: globalShardInformer.Informer().GetIndexer(),
+		globalShardLister:  globalShardInformer.Lister(),
+
+		globalWorkspaceTypeIndexer: globalWorkspaceTypeInformer.Informer().GetIndexer(),
+		globalWorkspaceTypeLister:  globalWorkspaceTypeInformer.Lister(),
+
+		logicalClusterIndexer: logicalClusterInformer.Informer().GetIndexer(),
+		logicalClusterLister:  logicalClusterInformer.Lister(),
+
+		discoveringDynamicSharedInformerFactory: discoveringDynamicSharedInformerFactory,
+
+		commit: committer.NewCommitter[*tenancyv1alpha1.Workspace, tenancyv1alpha1client.WorkspaceInterface, *tenancyv1alpha1.WorkspaceSpec, *tenancyv1alpha1.WorkspaceStatus](kcpClusterClient.TenancyV1alpha1().Workspaces()),
+	}
```

Comment:
lots of things in here are not used, are they? Can you prune it?

Reply:
Same below.

Reply:
I dont know how I managed to push this state. This was a total different version from what  I intended to push O_O